index.js

```diff
@@ -77,18 +77,21 @@ function extractJSONLDfromHTML(url, html) {
  * If dataCatalogUrl is not supplied, the default OA Data Catalog (https://openactive.io/data-catalogs/data-catalog-collection.jsonld) is used.
  *
  * @param {string} [dataCatalogUrl]
+ * @returns {Promise<{jsonld: Record<string,any>[], errors: string[]}>}
+ *
  */
 async function getAllDatasets(dataCatalogUrl = 'https://openactive.io/data-catalogs/data-catalog-collection.jsonld') {
   // Get Dataset URLs
   const datasetUrls = await getAllDatasetSiteUrls(dataCatalogUrl);
 
+  const errors = [];
   const jsonldFromDatasetUrls = (await Promise.all(datasetUrls.map(async (datasetUrl) => {
     let dataset;
     try {
       // Get JSONLD from dataset URLs
       dataset = (await axios.get(datasetUrl)).data;
     } catch (error) {
-      console.error(`getAllDatasets() - ${datasetUrl} could not be fetched`);
+      errors.push(`getAllDatasets() - ${datasetUrl} could not be fetched, ${error.response?.status ? `status:${error.response?.status}, ` : ''}message: "${error.message}".`);
```

Comment:
Perhaps better to structure this output, to be more useful to the user:

```suggestion
      errors.push({
        url: datasetUrl,
        status: error.response?.status,
        message: error.message,
      });
```
```